### PR TITLE
Add a portable v5.0 CodeAnalysis build task

### PR DESCRIPTION
**Pull request by agocke**
Moves the build task code to a shared project that we use to build
at net45 and a portable50 CodeAnalysis MSBuild task. Also upgrades to
nuget v3.2.1-rc since this is necessary to understand the dotnet4.3 spec
used in the project.json.

### CODE REVIEW

**Comment by agocke** [comment]
@dotnet/roslyn-compiler @jaredpar @jasonmalinowski FYI

**Comment by agocke** [comment]
@jaredpar You beat me to it so I'll merge around your PR

File: src/Compilers/Core/MSBuildTask/Portable/MSBuildTask.csproj [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  <ImportGroup Label="Settings">
+    <Import Project="..\..\..\..\..\build\Targets\VSL.Settings.targets" />
+  </ImportGroup>
+  <PropertyGroup>
+    <MinimumVisualStudioVersion>14.0</MinimumVisualStudioVersion>
+    <Configuration Condition=" '$(Configuration)' == '' ">Debug</Configuration>
+    <Platform Condition=" '$(Platform)' == '' ">AnyCPU</Platform>
+    <PlatformTarget>AnyCPU</PlatformTarget>
+    <ProjectGuid>{7AD4FE65-9A30-41A6-8004-AA8F89BCB7F3}</ProjectGuid>
+    <OutDir>$(OutDir)BuildTaskCodeAnalysis\</OutDir>
```
Comment (jaredpar) [inline src/Compilers/Core/MSBuildTask/Portable/MSBuildTask.csproj]: Why are we using a custom output directory here?

File: src/Compilers/Core/MSBuildTask/Portable/Properties/AssemblyInfo.cs [hunk 2]
```diff
@@ -0,0 +1,30 @@
+﻿using System.Resources;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+
+// General Information about an assembly is controlled through the following 
+// set of attributes. Change these attribute values to modify the information
+// associated with an assembly.
+[assembly: AssemblyTitle("MSBuildTask")]
```
Comment (jaredpar) [inline src/Compilers/Core/MSBuildTask/Portable/Properties/AssemblyInfo.cs]: Should we give this a name  with Microsoft.CodeAnalysis in it?

File: src/Compilers/Core/MSBuildTask/Shared/Csc.cs [hunk 3]
```diff
@@ -485,7 +485,7 @@ private bool InitializeHostCompiler(ICscHostObject cscHostObject)
                     CheckHostObjectSupport(param = nameof(Analyzers), analyzerHostObject.SetAnalyzers(Analyzers));
                 }
             }
-            catch (Exception e) when (!Utilities.IsCriticalException(e))
```
Comment (jaredpar) [inline src/Compilers/Core/MSBuildTask/Shared/Csc.cs]: Yes!!!

File: src/Compilers/Server/VBCSCompilerTests/VBCSCompilerTests.csproj [hunk 4]
```diff
@@ -42,9 +42,9 @@
       <Project>{F7712928-1175-47B3-8819-EE086753DEE2}</Project>
       <Name>TestUtilities.FX45</Name>
     </ProjectReference>
-    <ProjectReference Include="..\..\Core\MSBuildTask\MSBuildTask.csproj">
+    <ProjectReference Include="..\..\Core\MSBuildTask\Desktop\MSBuildTask.Desktop.csproj">
```
Comment (jaredpar) [inline src/Compilers/Server/VBCSCompilerTests/VBCSCompilerTests.csproj]: It's so weird that we have this dependency.  I'm going to try and remove it.

**Comment by jaredpar** [comment]
How did you end up solving the resource problem?

**Comment by agocke** [comment]
@jaredpar It works OK, you just have to be very careful to keep everything in the shared project.

File: src/VisualStudio/Setup/VisualStudioSetup.csproj [hunk 5]
```diff
@@ -23,6 +23,10 @@
     <CopyNuGetImplementations>true</CopyNuGetImplementations>
   </PropertyGroup>
   <ItemGroup Label="Project References">
+    <ProjectReference Include="..\..\Compilers\Core\MSBuildTask\Desktop\MSBuildTask.Desktop.csproj">
```
Comment (jasonmalinowski) [inline src/VisualStudio/Setup/VisualStudioSetup.csproj]: As observed yesterday, this shouldn't be here. What's the plan to remove it?

Comment (agocke) [inline src/VisualStudio/Setup/VisualStudioSetup.csproj]: I will remove it ASAP.

**Pull request merged**